lib/server/services/log/LogService.js

```diff
@@ -47,6 +49,36 @@ class LogService {
         return log ? logAdapter.toEntity(log) : null;
     }
 
+    /**
+     * Find and return all root logs
+     * @param {Partial<GetAllLogsDto>} dto The GetAllLogs DTO which contains all request data.
+     * @return {Promise<[Log]|null>} resolve with the logs found or null
+     */
+    async getAllRootLogs(dto = {}) {
+        const { count, logIds } = await getRootLogIds((queryBuilder) => {
+            const { query = {} } = dto;
+            const { page = {} } = query;
+
+            queryBuilder.include('user')
+                .include('tags')
+                .include({ association: 'runs', attributes: ['id', 'runNumber'] })
+                .include({ association: 'lhcFills', attributes: ['fillNumber'] })
+                .include('subsystems')
+                .include('attachments')
+                .include({ association: 'environments', attributes: ['id'] });
```

Comment:
If you extract log ids only, you probably don't need any include here.

Reply:
done